configurationPanel.js

```diff
@@ -0,0 +1,90 @@
+class ConfigurationPanel extends Panel
+{
+constructor(application,db)
+{
+    super(application);
+    this.name = "ConfigurationPanel";
+    this.db = db;
+}
+
+    show(root)
+    {
+        var template = document.querySelector("#ConfigurationPage"); //w przyszlosci pozbierac template do jednego obiektu
+        var templateClone = document.importNode(template.content, true);
+        var buttonC = templateClone.querySelector("[data-name='ChartPanelButton']");
+        var buttonA = templateClone.querySelector("[data-name='StartPanelButton']");
+        var addTeamButton = templateClone.querySelector("[data-name='AddTeamButton']");
+        root.appendChild(templateClone);
+        var loader = document.querySelector("[data-name='loader']")
+        loader.classList.remove("loading");
+        this.refreshTable(root);
+        buttonC.addEventListener("click", event => 
+    {
+        this.app.sendAction("ChartPanelRequested")
+    })
+        buttonA.addEventListener("click", event =>
+    {
+        this.app.sendAction("StartPanelRequested")
+    })
+    addTeamButton.addEventListener("click", event =>
+    {
+        var promise = this.db.addTeamToDb(root.querySelector("input").value);
+        promise = promise.then(() =>
+        {
+        this.refreshTable(root);
+        })
+    })
+    }
+    refreshTable(root,loader)
+    {
+        var loader = root.querySelector("[data-name='loader']")
+        loader.classList.add("loading");
+        var teamList = this.db.getTeamsFromDb();
+        teamList = teamList.then(teamList =>
+        {
+        teamList = JSON.parse(teamList)
```

Comment:
czemu panel sam sobie deserializuje dane? Nie powinien otrzymywać z bazy gotowych obiektów moduelu?

Reply:
Bardzo chętnie tylko ta wartość jest zwracana z promisa. Jak mam ją deserializować skoro to jest promise?